tests/test_cases/test_cocotb/test_concurrency_primitives.py

```diff
@@ -11,68 +11,63 @@
 
 
 @cocotb.test()
-def test_yield_list_stale(dut):
-    """ Test that a trigger yielded as part of a list can't cause a spurious wakeup """
+async def test_await_list_stale(dut):
+    """ Test that a trigger awaited as part of a list can't cause a spurious wakeup """
```

Comment:
Should probably adjust the test name now that there's no list here.

Reply:
There is still the list of ``waiters`` which I'd assume the "list" referred to?
If not, what would the docstring now be? I also don't completely get the meaning of "stale" in the test name.

Reply:
@garmin-mjames IIRC the docstring was from you initially, do you have a suggestion?

Reply:
This test is by @eric-wieser from https://github.com/cocotb/cocotb/issues/839#issuecomment-474171169, it was just related to an issue I filed at the same time.

The test is about two coroutines waiting on a shared trigger, but one of the coroutines is waiting on a list/`First` that contains the shared trigger.
If a different trigger in the `First` fires, we want to test that the scheduler removes that coro from the other still-pending trigger(s) so that it doesn't get woken up again later. (So, `events[1].set()` shouldn't resume `wait_for_lists`)

All that said, maybe:
```suggestion
    async def test_unfired_first_triggers(dut):
    """ Test that un-fired trigger(s) in First don't later cause a spurious wakeup """
```